packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,438 @@
+import {
+  channelIdRegex,
+  coerceChain,
+  swapRequestId,
+  txHashRegex,
+  estimateSwapDuration,
+  isEgressableSwap,
+  failedSwapMessage,
+} from '../../utils/swap';
+import { asyncHandler } from '../common';
+import { Failure } from '../swap';
+
+const router = express.Router();
+
+export enum StateV2 {
+  Failed = 'FAILED',
+  Complete = 'COMPLETE',
+  Broadcasted = 'SENT',
+  EgressScheduled = 'SENDING',
+  Swapping = 'SWAPPING',
+  DepositReceived = 'DEPOSIT_RECEIVED',
+  AwaitingDeposit = 'AWAITING_DEPOSIT',
+}
+
+const depositChannelInclude = {
+  failedBoosts: true,
+  failedSwaps: true,
+  affiliates: {
+    select: {
+      account: true,
+      commissionBps: true,
+    },
+  },
+} as const;
+
+const swapRequestInclude = {
+  swaps: { include: { fees: true }, orderBy: { nativeId: 'desc' } },
+  egress: { include: { broadcast: true } },
+  refundEgress: { include: { broadcast: true } },
+  fees: true,
+  ignoredEgresses: true,
+  swapDepositChannel: {
+    include: depositChannelInclude,
+  },
+} as const;
+
+const getSwapFields = (swap: Swap & { fees: SwapFee[] }) => ({
+  swapInputAmount: swap.swapInputAmount.toString(),
+  swapOutputAmount: swap.swapOutputAmount?.toString(),
+  scheduledAt: swap.swapScheduledAt,
+  scheduledBlockIndex: swap.swapScheduledBlockIndex ?? undefined,
+  executedAt: swap.swapExecutedAt,
+  executedBlockIndex: swap.swapExecutedBlockIndex ?? undefined,
+  retryCount: swap.retryCount,
+  latestSwapRescheduledAt: swap.latestSwapRescheduledAt,
+  latestSwapRescheduledBlockIndex: swap.latestSwapRescheduledBlockIndex ?? undefined,
+  fees: swap.fees.map((fee) => ({
+    type: fee.type,
+    ...getAssetAndChain(fee.asset),
+    amount: fee.amount.toString(),
+  })),
+});
+
+const getEgressStatusFields = (
+  egress: Egress | null | undefined,
+  broadcast: Broadcast | null | undefined,
+  ignoredEgresses: IgnoredEgress[] | undefined,
+  type: IgnoredEgress['type'] | undefined,
+  egressTrackerTxRef: string | null | undefined,
+  failedSwap: FailedSwap | null | undefined,
+) => {
+  const ignoredEgress = ignoredEgresses?.find((e) => e.type === type);
+  const failedAt =
+    failedSwap?.failedAt ?? broadcast?.abortedAt?.valueOf() ?? ignoredEgress?.ignoredAt?.valueOf();
+  const failedAtBlockIndex =
+    failedSwap?.failedBlockIndex ??
+    broadcast?.abortedBlockIndex ??
+    ignoredEgress?.ignoredBlockIndex;
+
+  return {
+    ...(egress && {
+      outputAmount: egress.amount?.toFixed(),
+      scheduledAt: egress.scheduledAt?.valueOf(),
+      scheduledBlockIndex: egress.scheduledBlockIndex ?? undefined,
+    }),
+    ...(broadcast && {
+      sentAt: broadcast?.succeededAt?.valueOf(),
+      sentAtBlockIndex: broadcast?.succeededBlockIndex ?? undefined,
+      sentTxRef: broadcast?.transactionRef ?? egressTrackerTxRef,
+    }),
+    ...(failedAt && failedAtBlockIndex && { failedAt, failedAtBlockIndex }),
+    ...(ignoredEgress && { ignoredAmount: ignoredEgress.amount?.toFixed() }),
+  };
+};
```

Comment:
renamed some of these fields to make it more human readable. I've removed `broadcast.requestedAt` since its basically the same as `egressScheduledAt`